version: "3.5"

services:
  server:
    image: my_server_image
    build:
#      context: ./django
      dockerfile: ./django/Dockerfile
    volumes:
      - ./django:/app
      - ./temp:/app/temp
      - static:/static
    ports:
      - "8000:8000"
    entrypoint: ["sh", "/app/entrypoint.sh"]
#    command: >
#      bash -c "python manage.py migrate && python manage.py migrate &&
#      python manage.py runserver 0.0.0.0:8000"

  rabbitmq:
    hostname: rabbitmq
    image: rabbitmq:3.9.7
    environment:
      - RABBITMQ_DEFAULT_USER=admin
      - RABBITMQ_DEFAULT_PASS=mypass
    ports:
      - "5672:5672"
    restart: on-failure
    logging:
      driver: none

  celery:
    hostname: celery
    image: my_server_image
#    build:
#      context: .
#      dockerfile: django/Dockerfile
    command: celery -A to_do_list worker -l info
    volumes:
      - ./django:/app
    depends_on:
      - server
      - rabbitmq
  nginx:
    build: ./nginx
    volumes:
      - static:/static
    ports:
      - "80:80"
    depends_on:
      - server

volumes:
  static:
